hcloud/networks/client.py

```diff
@@ -229,48 +230,68 @@ def create(
         :param routes: List[:class:`NetworkRoute <hcloud.networks.domain.NetworkRoute>`]
                 Array of routes set in this network
+        :param expose_routes_to_vswitch: Optional[bool]
+                Indicates if the routes from this network should be exposed to the vSwitch connection.
+                The exposing only takes effect if a vSwitch connection is active.
         :param labels: Dict[str, str] (optional)
                 User-defined labels (key-value pairs)
         :return: :class:`BoundNetwork <hcloud.networks.client.BoundNetwork>`
         """
         data = {"name": name, "ip_range": ip_range}
         if subnets is not None:
-            data["subnets"] = [
-                {
+            data_subnets = []
+            for subnet in subnets:
+                data_subnet = {
                     "type": subnet.type,
                     "ip_range": subnet.ip_range,
                     "network_zone": subnet.network_zone,
                 }
-                for subnet in subnets
-            ]
+                if subnet.vswitch_id is not None:
+                    data_subnet["vswitch_id"] = subnet.vswitch_id
+
+                data_subnets.append(data_subnet)
+            data["subnets"] = data_subnets
+
         if routes is not None:
             data["routes"] = [
                 {"destination": route.destination, "gateway": route.gateway}
                 for route in routes
             ]
+
+        if expose_routes_to_vswitch is not None:
+            data["expose_routes_to_vswitch"] = expose_routes_to_vswitch
+
         if labels is not None:
             data["labels"] = labels
 
         response = self._client.request(url="/networks", method="POST", json=data)
 
         return BoundNetwork(self, response["network"])
 
-    def update(self, network, name=None, labels=None):
+    def update(self, network, name=None, expose_routes_to_vswitch=None, labels=None):
         # type:(Network,  Optional[str],  Optional[Dict[str, str]]) -> BoundNetwork
         """Updates a network. You can update a network’s name and a network’s labels.
 
         :param network: :class:`BoundNetwork <hcloud.networks.client.BoundNetwork>` or :class:`Network <hcloud.networks.domain.Network>`
         :param name: str (optional)
                New name to set
+        :param expose_routes_to_vswitch: Optional[bool]
+                Indicates if the routes from this network should be exposed to the vSwitch connection.
+                The exposing only takes effect if a vSwitch connection is active.
         :param labels: Dict[str, str] (optional)
                User-defined labels (key-value pairs)
         :return: :class:`BoundNetwork <hcloud.networks.client.BoundNetwork>`
         """
         data = {}
         if name is not None:
             data.update({"name": name})
+
+        if expose_routes_to_vswitch is not None:
+            data["expose_routes_to_vswitch"] = expose_routes_to_vswitch
```

Comment:
The surrounding lines are using `data.update(dict)`, is there a reason to prefer one over another?

Reply:
The existing method is somewhat overkill, you have to create a new dict with the value you want to insert, and merge it into the existing dict, probably by iterating over incoming data.

The method I used is simply adding a value at the defined key.

I can use the existing method, this doesn't have a huge impact.